crates/pairwise_nostd_internal/src/misc.rs

```diff
@@ -160,4 +277,78 @@ mod tests {
             )
         }
     }
+
+    // maps between a 3D & 1D index
+    struct IdxMappingPair([isize; 3], isize);
```

Comment:
I would advocate for a tuple here, but it's not a big deal.

Reply:
I think I'm going to keep it. I kinda like it since we pass in an array of these instances to a helper test function (I think the nested braces and parentheses from a list of these tuples would be a little confusing). Plus, dropping this only save us the 2 highlighted lines (rustfmt won't let us save any more lines).

At the end of the day, I don't think it really matters since it's test-code